### PR TITLE
Decoder skeleton functions

### PR DESCRIPTION
**Pull request by Cvaztian**
Implemented skeleton code for decoder
Missing pixel drawing

### CODE REVIEW

File: decoder/decoder.c [hunk 1]
```diff
@@ -0,0 +1,86 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <stdint.h>
+#include <unistd.h>
+
+#define STB_IMAGE_WRITE_IMPLEMENTATION
+#include "stb_image/stb_image_write.h"
+
+struct Descriptor {
```
Comment (kevv87) [inline decoder/decoder.c:9]: Move to types

File: decoder/decoder.c [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+            int G = desc.encrypted_px & 0x00FF00;
+            int B = desc.encrypted_px & 0x0000FF;
+            //putpixel(px_counter%IMAGE_SIZE, px_counter/IMAGE_SIZE, desc.encrypted_px);        TODO: draw pixel
+            // Insert px RGB in image file
+            *px_iter = R;
+            *(px_iter + 1) = G;
+            *(px_iter + 2) = B;
+            px_counter += 1;
+        }
+        else { sleep(0.7); }
```
Comment (kevv87) [inline decoder/decoder.c:66]: Por que este sleep?

File: decoder/decoder.c [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+        is_run *= -1;
+    }
+    sleep(3);
```
Comment (kevv87) [inline decoder/decoder.c:69]: Sleep?

File: encoder/encoder.c [hunk 4]
```diff
@@ -57,6 +58,12 @@ int encrypt_pixel(int pixel, int key) {
     return pixel;
 }
 
+int format_hex_px(int red, int green, int blue) {
+    printf("%d, %d, %d\n", red, green, blue);
```
Comment (kevv87) [inline encoder/encoder.c:62]: If debug print, remove

File: encoder/encoder.c [hunk 5]
```diff
@@ -30,6 +30,7 @@ void read_image(char *file_name, struct ImgData *img_data) {
 	printf("Opening image file: %s...\n", file_name);
     img_data->img_ptr = stbi_load(file_name, &img_data->width, &img_data->height, &img_data->color_channels, 0);
 	if(img_data->img_ptr != NULL){
+        printf("%d, %d, %d\n\n\n", img_data->width, img_data->height, img_data->color_channels);
```
Comment (kevv87) [inline encoder/encoder.c:33]: Remove if its a debug print

File: encoder/encoder.c [hunk 6]
```diff
@@ -84,7 +91,7 @@ void insert_descriptor(struct Descriptor *desc, int pos) {
     strncpy(desc->insertion_time, time_str, 9);
     //(desc_array+(pos*sizeof(desc))) = desc;
     desc_array[pos] = *desc;
-    printf("Encrypted px: %d, Structure index: %d, Px position in file: %d, Insertion time: %s, Is read: %d\n",
+    printf("Encrypted px: %x, Structure index: %d, Px position in file: %d, Insertion time: %s, Is read: %d\n",
```
Comment (kevv87) [inline encoder/encoder.c:94]: remove

**Review by kevv87 — CHANGES_REQUESTED** [review]

File: encoder/encoder.c (hunk 6 above)
Comment (Cvaztian) [reply encoder/encoder.c:94]: Mama3.1415chas

File: encoder/encoder.c (hunk 5 above)
Comment (Cvaztian) [reply encoder/encoder.c:33]: Mama3.1415chas

File: encoder/encoder.c (hunk 4 above)
Comment (Cvaztian) [reply encoder/encoder.c:62]: Mama3.1415chas

File: decoder/decoder.c (hunk 3 above)
Comment (Cvaztian) [reply decoder/decoder.c:69]: Mama3.1415chas

File: decoder/decoder.c (hunk 2 above)
Comment (Cvaztian) [reply decoder/decoder.c:66]: Mama3.1415chas

File: decoder/decoder.c [hunk 7]
```diff
@@ -0,0 +1,208 @@
...
+    // for testing GTK      TODO: remove for
+    for (int i=0; i<500; i++) {
+        common_array[i] = 1;
+        sleep(1);
+    }
+
+    unsigned char *px_iter;
+    int px_counter = 0;
+    int is_run = 1;
+    /*while(px_counter < decod_cfg.img_size) {
```
Comment (kevv87) [inline decoder/decoder.c:115]: Esto lo borra? O lo dejamos asi?

File: encoder/encoder.c (hunk 5 above)
Comment (kevv87) [reply encoder/encoder.c:33]: xD Lo quitamos o que?

**Review by kevv87 — CHANGES_REQUESTED** [review]

**Review by kevv87 — APPROVED** [review]

**Pull request merged**